src/main/java/mate/academy/springboot/datajpa/repository/specification/product/ProductCategorySpecification.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.springboot.datajpa.repository.specification.product;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.Join;
+import jakarta.persistence.criteria.JoinType;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.specification.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductCategorySpecification implements SpecificationProvider<Product> {
+    private static final String FILTER_KEY = "categoryIn";
+    private static final String FIELD_NAME = "category";
+
+    @Override
+    public Specification<Product> getSpecification(String[] categories) {
+        return ((root, query, criteriaBuilder) -> {
+            Join<Product, Category> join = root.join("category", JoinType.INNER);
```

Comment:
```suggestion
            Join<Product, Category> join = root.join("category", JoinType.INNER);
```
"category" also better makes constant